universe/Universe.cpp

```diff
@@ -1068,10 +1070,19 @@ namespace {
             target_objects.insert(target_objects.end(), target_set->begin(), target_set->end());
         }
 
+        // log condition scope matches, except for Source
+        if (cond->Description() != " that is the source object") {  // TODO: derive this comparison text
```

Comment:
could `dynamic_cast` to a pointer to a `Source` condition, and check that the cast worked, rather than testing text